src/core/io/writer/h5md_core.cpp

```diff
@@ -47,562 +52,340 @@ static void backup_file(const std::string &from, const std::string &to) {
+
+  write_td_particle_property<2>(
+      prefix, n_part_global, particles, datasets["particles/atoms/id/value"],
+      [](auto const &p) { return Utils::Vector<int, 1>{p.p.identity}; });
+  write_dataset(Utils::Vector<double, 1>{time},
+                datasets["particles/atoms/id/time"], change_extent_1d,
+
+                offset_1d, count_1d);
+  write_dataset(Utils::Vector<int, 1>{step},
+                datasets["particles/atoms/id/step"], change_extent_1d,
+                offset_1d, count_1d);
+
+  write_td_particle_property<2>(
+      prefix, n_part_global, particles,
+      datasets["particles/atoms/species/value"],
+      [](auto const &p) { return Utils::Vector<int, 1>{p.p.type}; });
+
+  write_td_particle_property<2>(
+      prefix, n_part_global, particles, datasets["particles/atoms/mass/value"],
+      [](auto const &p) { return Utils::Vector<double, 1>{p.p.mass}; });
+
+  write_td_particle_property<3>(
+      prefix, n_part_global, particles,
+      datasets["particles/atoms/position/value"],
+      [&](auto const &p) { return folded_position(p.r.p, box_geo); });
+  write_td_particle_property<3>(prefix, n_part_global, particles,
+                                datasets["particles/atoms/image/value"],
+                                [](auto const &p) { return p.l.i; });
+
+  write_td_particle_property<3>(prefix, n_part_global, particles,
+                                datasets["particles/atoms/velocity/value"],
+                                [](auto const &p) { return p.m.v; });
+
+  write_td_particle_property<3>(prefix, n_part_global, particles,
+                                datasets["particles/atoms/force/value"],
+                                [](auto const &p) { return p.f.f; });
+  write_td_particle_property<2>(
+      prefix, n_part_global, particles,
+      datasets["particles/atoms/charge/value"],
+      [](auto const &p) { return Utils::Vector<double, 1>{p.p.q}; });
 }
+void File::write_connectivity(const ParticleRange &particles) {
+  MultiArray3i bond(boost::extents[0][0][0]);
+  int particle_index = 0;
+  for (auto const &p : particles) {
+    int nbonds_local = bond.shape()[1];
+    for (auto const b : p.bonds()) {
+      auto const partner_ids = b.partner_ids();
+      if (partner_ids.size() == 1) {
+        bond.resize(boost::extents[1][nbonds_local + 1][2]);
+        bond[0][nbonds_local][0] = p.p.identity;
+        bond[0][nbonds_local][1] = partner_ids[0];
+        nbonds_local++;
+      }
+    }
+    particle_index++;
+  }
 
-void File::Flush() {
-  if (m_write_ordered) {
-    if (this_node == 0)
-      H5Fflush(m_h5md_file.hid(), H5F_SCOPE_GLOBAL);
-  } else
-    H5Fflush(m_h5md_file.hid(), H5F_SCOPE_GLOBAL);
+  int n_bonds_local = bond.shape()[1];
+  int n_bonds_total = n_bonds_local;
+  int prefix_bonds = 0;
+  MPI_Exscan(&n_bonds_local, &prefix_bonds, 1, MPI_INT, MPI_SUM, m_comm);
+  MPI_Allreduce(&n_bonds_local, &n_bonds_total, 1, MPI_INT, MPI_SUM, m_comm);
```

Comment:
Please use boost.mpi, it has an overload which returns the accumulated value, so it can be const like so
```c++
auto const n_bond_total = boost::mpi::all_reduce(...);
```